lib/jobs/pdu-node-relations.js

```diff
@@ -0,0 +1,142 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = pduRelationsJobFactory;
+di.annotate(pduRelationsJobFactory, new di.Provide('Job.Catalog.PduRelations'));
+di.annotate(pduRelationsJobFactory, new di.Inject(
+    'Job.Base',
+    'Services.Waterline',
+    'Logger',
+    'Util',
+    'Promise',
+    'Assert'
+));
+
+function pduRelationsJobFactory(
+    BaseJob,
+    waterline,
+    Logger,
+    util,
+    Promise,
+    assert
+) {
+
+    var logger = Logger.initialize(pduRelationsJobFactory);
+
+    /**
+     *
+     * @param {Object} [options]
+     * @constructor
+     */
+    function PduRelationsJob(options, context, taskId) {
+        PduRelationsJob.super_.call(this, logger, options, context, taskId);
+
+        this.nodeId = context.target || options.nodeId;
+        assert.isMongoId(this.nodeId);
+    }
+
+    util.inherits(PduRelationsJob, BaseJob);
+
+    /**
+     * Globals
+     */
+
+    var RELATION_TYPE = 'poweredBy';
+
+    /**
+     * @memberOf PduRelationsJob
+     */
+    PduRelationsJob.prototype._run = function run() {
+        var self = this;
+        return waterline.nodes.findByIdentifier(self.nodeId)
```

Comment:
there is a check for returned value in needByIdentifier, it's 'node' here, if don't use needByIdentifier , need a check before 'node.type' != 'pdu',  otherwise, if there's no 'node' found by 'nodeId'  there will be error like 'TypeError: Cannot read property 'type' of undefined'